src/main/java/com/amazon/corretto/hotpatch/patch/impl/set/PatchSetPatcher.java

```diff
@@ -0,0 +1,177 @@
+    public int install(final Map<String, String> args, final int asmApiVersion, final Instrumentation instrumentation,
+                       final Logger logger, final boolean staticAgent) {
+        this.instrumentation = instrumentation;
+        this.logger = logger;
+
+        List<ClassTransformerHotPatch> patches = getEnabledPatches(args);
+        this.transformer = new PatchSetTransformer(patches, asmApiVersion, logger);
+
+        if (staticAgent) {
+            // As we are being installed during startup, we don't care about retransforming. The transformer will
+            // patch the classes as they are being loaded.
+            instrumentation.addTransformer(transformer);
+        } else {
+            // We are being attached to a running VM, we need to retransform the classes targeted by the patches that
+            // have already been loaded.
+            instrumentation.addTransformer(transformer, true);
+
+            boolean patchesApplied = false;
+            for (Class<?> c : instrumentation.getAllLoadedClasses()) {
+                String className = c.getName();
+                for (ClassTransformerHotPatch patch : patches) {
+                    if (patch.isTargetClass(className)) {
+                        logger.log("Patching + " + className + " (" + c.getClassLoader() + ") with patch "
+                                + patch.getName());
+                        try {
+                            instrumentation.retransformClasses(c);
+                            patchesApplied = true;
+                        } catch (UnmodifiableClassException uce) {
+                            logger.log(String.valueOf(uce));
+                        }
+                    }
+                }
+            }
+
+            if (patchesApplied) {
+                logger.log("Vulnerable classes were not found. This agent will continue to run " +
+                        "and transform the target classes if they are loaded. Note that if you have shaded " +
+                        "or otherwise changed the package name for target classes, then this tool may not " +
+                        "find them.");
+            }
+        }
+
+        return Patcher.SUCCESS;
+    }
+
+    @Override
+    public int uninstall() {
+        instrumentation.removeTransformer(transformer);
+        // Retransform after we've removed the transformer to restore the initial class versions.
+        for (Class<?> c : instrumentation.getAllLoadedClasses()) {
+            String className = c.getName();
+            for (ClassTransformerHotPatch patch : getPatches()) {
+                if (patch.isTargetClass(className)) {
+                    logger.log("Un-Patching " + c + " (" + c.getClassLoader() + ") of " + patch.getName());
+                    try {
+                        instrumentation.retransformClasses(c);
+                    } catch (UnmodifiableClassException uce) {
+                        logger.log(String.valueOf(uce));
```

Comment:
Same, this method always returns success

Reply:
Same